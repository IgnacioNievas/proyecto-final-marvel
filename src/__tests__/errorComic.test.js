import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import ErrComic from '../components/erroresComponent/errorComic';

test('<ErrComic/> , prueba de existencia de elementos', () => {
	const wrapper = render(<ErrComic />);
	wrapper.debug();
	const h1 = screen.getByTestId('titulo');
	expect(h1.tagName).toBe('H1');
	expect(h1.tagName).not.toBe('SPAN', 'H2', 'H3', 'H4', 'P');
	expect(h1).toHaveTextContent('Comic no encontrado');
	expect(h1).toBeInTheDocument();

	const h2 = screen.getByTestId('titulo2');
	expect(h2.tagName).toBe('H2');
	expect(h2.tagName).not.toBe('SPAN', 'H1', 'H3', 'H4', 'P', 'INPUT');
	expect(h2).toHaveTextContent(
		'Tal vez no exista o no se encuentre en este personaje'
	);
	expect(h2).toBeInTheDocument();

	const errComic = renderer.create(<ErrComic />).toJSON();

	expect(errComic).toMatchSnapshot();
});
