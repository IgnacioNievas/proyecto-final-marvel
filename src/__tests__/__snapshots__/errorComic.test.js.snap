// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrComic/> , prueba de existencia de elementos 1`] = `
<div
  className="sc-bdnxRM bjKOGz"
>
  <h1
    className="h"
    data-testid="titulo"
  >
    Comic no encontrado
    <i
      className="fas fa-exclamation-triangle"
    />
  </h1>
  <h2
    className="sub"
    data-testid="titulo2"
  >
    Tal vez no exista o no se encuentre en este personaje
  </h2>
  <h3
    className="sub3"
    data-testid="titulo3"
  >
    O thanos lo habrá con guantelete del infinito
  </h3>
  <p
    className="p2"
    data-testid="parrafo"
  >
    • Por favor haz click sobre para volver o en el logo para volver a los personajes
  </p>
  <p
    className="p2"
    data-testid="parrafo2"
  >
    • Sino borrar la palabra del buscador y volveras a los comics que elegiste
  </p>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="thanos"
      className="img2"
      data-testid="img"
      src="thanos.png"
    />
  </a>
</div>
`;
